dependencies/osx/install-dependencies-osx

```diff
@@ -22,6 +22,9 @@ then
   ./install-overlay
 fi
 
+# install postgres, needed by SOCI
+brew install postgres
```

Comment:
We don't formally have a dependency on homebrew for dev machines, although in practice I think we all use it. I'm fine with introducing one, but make sure you also update the wikis if you do, and maybe add check here to see if homebrew is installed (so attempting to set up a machine w/o homebrew will give a more useful error).